pkg/tfvars/openstack/bootstrap_ignition.go

```diff
@@ -137,6 +138,20 @@ func generateIgnitionShim(userCA string, clusterID string, bootstrapConfigURL st
 		},
 	}
 
+	var httpProxy, httpsProxy *string
+	if proxy.HTTPProxy != "" {
+		httpProxy = &proxy.HTTPProxy
+	}
+	if proxy.HTTPSProxy != "" {
+		httpsProxy = &proxy.HTTPSProxy
+	}
```

Comment:
If `httpProxy` and `httpsProxy` can be *string type, then there is actually no reason to check for nil. You might as well just set them to the pointer in the proxy object pointer. This is because the [zero value for a pointer in GO is always nil](https://yourbasic.org/golang/default-zero-value/). However, what does need to be checked here is whether the proxy pointer itself is nil, since you de-reference its values by checking `proxy.HTTPProxy != ""`.

Reply:
This is why the code panics on line 142 when the proxy is unset (nil)
```log
INFO[2021-04-07T17:10:24Z] panic: runtime error: invalid memory address or nil pointer dereference 
INFO[2021-04-07T17:10:24Z] [signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0xacf7fec] 
INFO[2021-04-07T17:10:24Z]                                              
INFO[2021-04-07T17:10:24Z] goroutine 1 [running]:                       
INFO[2021-04-07T17:10:24Z] github.com/openshift/installer/pkg/tfvars/openstack.generateIgnitionShim(0x0, 0x0, 0xc0007dab20, 0x14, 0xc000738360, 0x58, 0xc0012763c0, 0xb7, 0x0, 0x0, ...) 
INFO[2021-04-07T17:10:24Z] 	/go/src/github.com/openshift/installer/pkg/tfvars/openstack/bootstrap_ignition.go:142 +0x46c 
```

Reply:
The check for not empty **is** necessary. There is a distinction between a nil pointer and a pointer to an empty string. A pointer to an empty string will not be omitted from the marshaled ignition config.